Framework/DataHandling/src/SaveOpenGenieAscii.cpp

```diff
@@ -49,337 +46,382 @@ void SaveOpenGenieAscii::init() {
   std::vector<std::string> header{"ENGIN-X Format", "Basic Format"};
   declareProperty("OpenGenieFormat", "ENGIN-X Format",
                   boost::make_shared<Kernel::StringListValidator>(header),
-                  "The format required to succesfully load the file to "
+                  "The format required to successfully load the file to "
                   "OpenGnie: ENGIN-X Format (default)");
-  declareProperty(
-      "SpecNumberField", "",
-      "Spec number is a required field for ENGIN-X format of OpenGenie,"
-      "an example of spec number would be: 1201 - 1400");
-  setPropertySettings("SpecNumberField",
-                      make_unique<VisibleWhenProperty>(
-                          "OpenGenieFormat", IS_EQUAL_TO, "ENGIN-X Format"));
 }
 
+/**
+  * Writes an OpenGenie file in ASCII at the user specified path
+  */
 void SaveOpenGenieAscii::exec() {
-  // Process properties
-
   // Retrieve the input workspace
-  /// Workspace
-  ws = getProperty("InputWorkspace");
-  int nSpectra = static_cast<int>(ws->getNumberHistograms());
-  int nBins = static_cast<int>(ws->blocksize());
-
-  // Retrieve the filename from the properties
-  std::string filename = getProperty("Filename");
+  m_inputWS = getProperty("InputWorkspace");
 
-  // Output string variables
-  const std::string singleSpc = " ";
-  const std::string fourspc = "    ";
+  inputValidation();
 
-  // file
-  std::ofstream outfile(filename.c_str());
-  if (!outfile) {
-    g_log.error("Unable to create file: " + filename);
-    throw Exception::FileError("Unable to create file: ", filename);
-  }
-  if (nBins == 0 || nSpectra == 0)
-    throw std::runtime_error("Trying to save an empty workspace");
+  // Its better to reserve over and the number of logs gives us a good
+  // estimate as some logs are not included whilst some other params are.
+  m_outputVector.reserve(m_inputWS->run().getLogData().size());
 
-  // Axis alphabets
-  const std::string Alpha[] = {"\"e\"", "\"x\"", "\"y\""};
+  // Whilst this doesn't weigh in the processing
+  // required it breaks down the algorithm nicely for the moment
+  const int numOfSteps = 6;
+  Progress progressBar(this, 0, 1, numOfSteps);
 
-  // Get IncludeHeader property
-  const bool headers = getProperty("IncludeHeader");
-  // if true write file header
-  if (headers) {
-    writeFileHeader(outfile);
+  const std::string formatType = getProperty("OpenGenieFormat");
+  if (formatType == "ENGIN-X Format") {
+    progressBar.report("Generating ENGINX header");
+    applyEnginxFormat();
+  } else {
+    const std::string err_msg("Unrecognized format \"" + formatType + "\"");
+    throw std::runtime_error(err_msg);
```

Comment:
No previously if you tried to save in the basic format it would throw an exception [here](https://github.com/mantidproject/mantid/blob/master/Framework/DataHandling/src/SaveOpenGenieAscii.cpp#L115) hence I know for sure nobody is using the basic format.

The algorithm has never actually worked correctly since its initial release as it was missing some essential fields. However they don't use it very often as it is used when they are processing event workspaces as their bespoke routines handle everything with matrix workspaces. 
As they are gradually doing more with event workspaces and they have data waiting to be processed it became a priority when previously it wasn't an issue for them.